src/user.jl

```diff
@@ -310,7 +310,26 @@ function preferunits(u0::Units, u::Units...)
             "For instance, it should not be used with units of dimension 𝐋^2.")
         end
         y = typeof(dim).parameters[1][1]
-        promotion[name(y)] = typeof(unit).parameters[1][1]
+        promotion[name(y)] = unit
+    end
+
+    # Let's run a quick check for anything where, for instance, current was defined is C/ms and time
+    # was defined in ns. We'll give a warning if this is the case so the user knows they may get
+    # unexpected (though still correct) results. There may be cases where people would want to do this,
+    # so we won't stop them, just let them know they're doing it.
+    ulist = [([typeof(i[2]).parameters[1] for i in promotion]...)...]
+    ulist = [i^(1/i.power) for i in ulist]
```

Comment:
Similar to the line above, you can use a lazy generator instead of allocating a vector:
```suggestion
    ulist = (i^(1/i.power) for i in ulist)
```